cfme/services/requests.py

```diff
@@ -1,286 +1,370 @@
+        boot_disk_size = SummaryFormItem('Properties', 'Boot Disk Size ')
+        is_preemtible = Checkbox(name='hardware__is_preemptible')
+
+    @View.nested
+    class customize(Tab):  # noqa
+        username = SummaryFormItem('Credentials', 'Username')
+        ip_mode = SummaryFormItem('IP Address Information', 'Address Mode')
+        hostname = SummaryFormItem('IP Address Information', 'Address Mode')
+        subnet_mask = SummaryFormItem('IP Address Information', 'Subnet Mask')
+        gateway = SummaryFormItem('IP Address Information', 'Gateway')
+        dns_server_list = SummaryFormItem('DNS', 'DNS Server list')
+        dns_suffix_list = SummaryFormItem('DNS', 'DNS Suffix list')
+        subnet_mask = SummaryFormItem('IP Address Information', 'Subnet Mask')
+        customize_template = SummaryFormItem('Customize Template', 'Script Name')
+
+    @View.nested
+    class schedule(Tab):  # noqa
+        when_provision = SummaryFormItem('Schedule Info', 'When to Provision')
+        stateless = Checkbox(name='shedule__stateless')
+        power_on = SummaryFormItem('Lifespan', 'Power on virtual machines after creation')
+        retirement = SummaryFormItem('Lifespan', 'Time until Retirement')
+        retirement_warning = SummaryFormItem('Lifespan', 'Retirement Warning')
+
+    breadcrumb = BreadCrumb()
+    toolbar = RequestDetailsToolBar()
+    reason = Input(name='reason')
+    submit = Button(title='Submit')
+    cancel = Button(title="Cancel this provisioning request")
+
+    def approve(self, reason_text, cancel=False):
+        """Approve currently opened request
+        Args:
+            reason_text: Reason for approving the request.
+            cancel: Cancel Approval and move to all request list
+        """
+        self.toolbar.approve.click()
+        self.reason.fill(reason_text)
+        if not cancel:
+            self.submit.click(handle_alert=not cancel)
+        else:
+            self.breadcrumb.click_location(self.breadcrumb.locations[0], handle_alert=True)
+        self.flash.assert_no_error()
+
+    def deny(self, reason_text, cancel=False):
+        """Deny currently opened request
+        Args:
+            reason_text: Reason for denying the request.
+            cancel:  Cancel Denial and move to all request list
+        """
+        self.toolbar.deny.click()
+        self.reason.fill(reason_text)
+        if not cancel:
+            self.submit.click(handle_alert=not cancel)
+        else:
+            self.breadcrumb.click_location(self.breadcrumb.locations[0], handle_alert=True)
+        self.flash.assert_no_error()
+
+    def delete(self, cancel=False):
+        """Delete currently opened request
+        Args:
+            cancel: Whether to cancel the deletion.
+        """
+        self.toolbar.delete.click(handle_alert=not cancel)
+        self.flash.assert_no_error()
 
 
 @navigator.register(Request, 'All')
 class RequestAll(CFMENavigateStep):
+    VIEW = RequestView
```

Comment:
what about plural name since it displays you view with many requests ?
how do you think ?